ANTLR/BaseVisitor.py

```diff
@@ -39,6 +39,7 @@ def __init__(self):
         super().__init__()
         self.memory = dict()
         self.tmp_memory = dict()
+        self.returned_value_from_list_function = None
```

Comment:
I think we can use tmp_memory for this.

Ex.
```
If 'list_return_val' in self.tmp_memory:
     return self.tmp_memory['list_return_val']
```

Or even use the 'list' arg
Imagine if you use `list(1...4).map(*2).drop()`
After drop() we return a value so we cannot use another list func. Saving it to 'list' key, will cause that it will be returned

Reply:
done